### ISSUE TITLE
Read FLEx WS info from hg and store it as project metadata

### ISSUE THREAD

**Issue by myieye** [sillsdev/languageforge-lexbox#844]
> [!NOTE]  
> The scope of this issue is **only FLEx projects**.

We'd like to know what writing systems/languages are being used by the projects we host. That has value, because:
1) It shows us what projects might be related to each other or even redundant.
2) We can actually see what languages dictionaries are (not) being developed for. That could be relevant for many reasons e.g. related projects liks Bloom or Keyman or simply statistics.

Currently all we have is an unvalidated, human entered language tag that is embedded in the project code (e.g. `eng-flex`), but:
- That's not reliable
- FLEx projects always use at least two languages (1 analysis and 1 vernacular), but could target any number of languages

As far as I can tell, the WS tags are stored in `.hg\store\data\_general\_language_project.langproj.i` in the tags: `CurAnalysisWss`, `CurVernWss`, `AnalysisWss`, `VernWss`, `CurPronunWss` and `HomographWs`.

This is my basic understanding of/guesses about those tags:
- "Cur" refers to the WS's that are currently configured
- Non-Cur (e.g. `VernWss`) are WS's that are no longer configured (or maybe deselected - see screenshot), but might be in use (e.g. maybe embedded in examples are something 🤷)
![image](https://github.com/sillsdev/languageforge-lexbox/assets/12587509/e002708a-8f54-4201-bfeb-759e26c1005b)
- CurPronunWss and HomographWs were the same in the projects I looked at. My best guess is that they by default correlate to the default vernacular WS? But I dunno and I don't think that's important for this issue.

This is what I think is the rough approach for this task:
1) Extend the `ProjectService.UpdateProjectMetadata` method to also fetch this metadata
2) Fetch the metadata using a new command in `command-runner.sh` (e.g. `flex-writing-systems`). The new command will be similar to `lexentrycount`, but I think we probably want to return the whole XML contents and process it in .NET.
3) In .NET use some XML API to find the relevant tags and then
4) Store the WS info in a new JSON column in the FlexProjectMetadata table.
Maybe something like this:
```ts
[
{
  ws: 'lwl',
  vernacular?: true, // i.e. this propery only exists if it's a vernacular WS
  analysis?: true, // i.e. this propery only exists if it's an analysis WS
  current?: true, // i.e. this propery only exists if it's in a "cur" list
},
// ...
]
```

**Comment by megahirt** [comment]
@jasonleenaylor can you confirm that Tim's hunch about `CurVernWss `and `CurAnalysisWss `are correct?  I assume we are only interested in these two properties and not the others?

**Comment by rmunn** [comment]
My (not quite perfect) understanding of the FieldWorks data model is that Tim is correct. For example, if you look at https://github.com/sillsdev/FieldWorks/blob/release/9.1/TestLangProj/TestLangProj.fwdata lines 912233 to 912235, you'll see the following:

```xml
<AnalysisWss>
<Uni>en fr es de zh-CN fr-fonipa qaa-x-sta ar-IQ dv qaa-x-fwt qaa-x-sim qaa-x-got qaa-x-pig</Uni>
</AnalysisWss>
```

On lines 912321 to 912329:

```xml
<CurAnalysisWss>
<Uni>en fr</Uni>
</CurAnalysisWss>
<CurPronunWss>
<Uni>qaa-fonipa-x-kal</Uni>
</CurPronunWss>
<CurVernWss>
<Uni>qaa-x-kal qaa-fonipa-x-kal qaa-Zxxx-x-kal-audio</Uni>
</CurVernWss>
```

And on lines 912500 to 912502:

```xml
<VernWss>
<Uni>qaa-x-kal qaa-fonipa-x-kal qaa-Zxxx-x-kal-audio</Uni>
</VernWss>
```

If you dig through the data looking for the `qaa-x-pig` writing system (Pig Latin), you'll find that there are no fields tagged with `qaa-x-pig` as the writing system, but it *does* exist in text data: the definition that starts on line 503404 (with the text "This is the English definition for 2.") is tagged as `ws="en"` but contains several runs tagged with other writing systems, including Pig Latin.

**Comment by rmunn** [comment]
I agree with Tim's assessment that "current" writing systems means "user checked the box". I looked at the FwWritingSystemSetupModel.Save() method:

https://github.com/sillsdev/FieldWorks/blob/e56040b18712c046065a0303f5f378398e94892d/Src/FwCoreDlgs/FwWritingSystemSetupModel.cs#L604

And I read the code for RemoveWritingSystems() as well (scroll down to line 802). "Current" means checked in that dialog, and those will be displayed as data options in fields. "All" writing systems may contain data in the project (e.g., the Pig Latin writing system in TestLangProj), so any tool wanting to display the data (such as a dictionary viewer) will need to be able to look those writing systems up to get the right font, keyboards, collations, and so on. But if we're trying to figure out what language(s) a project is associated with, the "Current" WSes are what we want.

**Comment by rmunn** [comment]
Whoops: here's a comment I wrote a couple days ago but never actually hit the Comment button.

I agree with Tim's assessment that "current" writing systems means "user checked the box". I looked at the FwWritingSystemSetupModel.Save() method:

https://github.com/sillsdev/FieldWorks/blob/e56040b18712c046065a0303f5f378398e94892d/Src/FwCoreDlgs/FwWritingSystemSetupModel.cs#L604

And I read the code for RemoveWritingSystems() as well (scroll down to line 802). "Current" means checked in that dialog, and those will be displayed as data options in fields. "All" writing systems may contain data in the project (e.g., the Pig Latin writing system in TestLangProj), so any tool wanting to display the data (such as a dictionary viewer) will need to be able to look those writing systems up to get the right font, keyboards, collations, and so on. But if we're trying to figure out what language(s) a project is associated with, the "Current" WSes are what we want.
